File: apps/dotnet/FizzBuzz/Domain/Type/FizzBuzzType01.cs
namespace FizzBuzz.Domain.Type;

using FizzBuzz.Domain.Model;

public class FizzBuzzType01 : IFizzBuzzType
{
    private const int FizzNumber = 3;
    private const int BuzzNumber = 5;

    public FizzBuzzValue Generate(int number)
    {
        string value;
        if (IsFizzBuzz(number))
            value = "FizzBuzz";
        else if (IsFizz(number))
            value = "Fizz";
        else if (IsBuzz(number))
            value = "Buzz";
        else
            value = number.ToString();

        return new FizzBuzzValue(number, value);
    }

    private static bool IsFizz(int number) => number % FizzNumber == 0;
    private static bool IsBuzz(int number) => number % BuzzNumber == 0;
    private static bool IsFizzBuzz(int number) => IsFizz(number) && IsBuzz(number);
}

File: apps/node/src/fizzbuzz/domain/type/fizz-buzz-type-01.ts
import { FizzBuzzValue } from "../model/fizz-buzz-value.js";
import type { FizzBuzzType } from "./fizz-buzz-type.js";

export class FizzBuzzType01 implements FizzBuzzType {
  generate(number: number): FizzBuzzValue {
    if (number % 15 === 0) {
      return new FizzBuzzValue("FizzBuzz", number);
    }
    if (number % 3 === 0) {
      return new FizzBuzzValue("Fizz", number);
    }
    if (number % 5 === 0) {
      return new FizzBuzzValue("Buzz", number);
    }

    return new FizzBuzzValue(number.toString(), number);
  }
}
